include/openssl/pem.h

```diff
@@ -455,7 +457,7 @@ EVP_PKEY *PEM_read_bio_Parameters_ex(BIO *bp, EVP_PKEY **x,
 EVP_PKEY *PEM_read_bio_Parameters(BIO *bp, EVP_PKEY **x);
 int PEM_write_bio_Parameters(BIO *bp, const EVP_PKEY *x);
 
-# ifndef OPENSSL_NO_DSA
+# if !defined(OPENSSL_NO_DSA) || !defined(OPENSSL_NO_DEPRECATED_3_0)
```

Comment:
Break this into two condition lines.  OpenSSL::ParseC doesn't have such advanced expression processor that it handles this properly, so `util/mknum.pl` may fail to detect these feature indicators.

Reply:
Also, if you're deprecating these functions, please add the deprecation attribute.

Reply:
Done in a few places.